gradio/components.py

```diff
@@ -472,7 +477,7 @@ def preprocess_example(self, x: float | None) -> float | None:
         if x is None:
             return None
         else:
-            return float(x)
+            return int(x) if self.integer else float(x)
```

Comment:
Let's update the docstring and type hint.

Reply:
Done!